src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
```

Comment:
... because the other locations are about values and not the names of values. Think of the difference between the `Text` and `Value` properties of a `SelectListItem` or (equivalently) between the body of an `<option>` element and its `value` attribute.

Bottom line, don't make changes beyond the scope of the identified problem -- bodies of generated `<option>` elements and the `@Html.Display[For]()` return value.